streams/src/main/java/org/apache/kafka/streams/processor/TaskId.java

```diff
@@ -80,6 +83,35 @@ public String toString() {
         return namedTopology != null ? namedTopology + "_" + topicGroupId + "_" + partition : topicGroupId + "_" + partition;
     }
 
+    /**
+     * @throws TaskIdFormatException if the taskIdStr is not a valid {@link TaskId}
+     */
+    public static TaskId parse(final String taskIdStr) {
+        final int firstIndex = taskIdStr.indexOf('_');
```

Comment:
I guess I missed the "name topology" change. What is a topology name? How to set it? And do we ensure that we don't allow `_` in its name?

Reply:
Yes, we plan to restrict the `_` character. If we want to loosen that up later we can just parse this from the back, but I think it's reasonable to just disallow `_` completely. 

> What is a topology name?

Great question. Not necessarily a short answer but I can try -- basically an independent and isolated piece of a topology that can be added/removed/etc at will, even on a running app. 

> How to set it?

The skeleton API was merged in [#10615](https://github.com/apache/kafka/pull/10615/files), it has/is evolving a bit since then but the basic idea holds -- each NamedTopology is built up with a special builder called the NamedTopologyStreamsBuilder. And a dedicated KafkaStreams wrapper is the entry point for starting up an app using NamedTopologies. All currently under the `internals` package while it's under the experimental phase so it should not be possible for a user to end up with anything NamedTopology through public APIs.